bl31/bl31.mk

```diff
@@ -12,6 +12,7 @@ BL31_SOURCES		+=	bl31/bl31_main.c				\
 				bl31/aarch64/runtime_exceptions.S		\
 				bl31/aarch64/crash_reporting.S			\
 				bl31/bl31_context_mgmt.c			\
+				bl31/plat_params.c				\
```

Comment:
Yeah, it could, although I feel that would just add extra busy-work for platforms trying to use it. Do you think building it by default would be an issue? Note that this does add no extra bytes to the binary if the platform doesn't use it, since Trusted Firmware is linked with --gc-sections and will garbage-collect all unused code and data.

I can put it in common/, but I felt that bl31/ was the more appropriate place since it only makes sense for use in BL31 (whereas common/ seems to contain code that is useful in more than one BL).